be1-go/hub/hub/message_handling.go

```diff
@@ -0,0 +1,287 @@
+package hub
```

Comment:
It seems weird to have to packages named hub, one inside the other: `hub/hub`. Maybe you can call the second package `general`